src/Tokenizer.js

```diff
@@ -194,134 +194,136 @@ module.exports = class Tokenizer {
   }
 
   list(src) {
-    const cap = this.rules.block.list.exec(src);
+    let cap = this.rules.block.list.exec(src);
     if (cap) {
-      let raw = cap[0];
-      const bull = cap[2];
+      let raw, istask, ischecked, indent, i, blankLine, endsWithBlankLine,
+        line, lines, itemContents;
+
+      let bull = cap[1].trim();
       const isordered = bull.length > 1;
 
       const list = {
         type: 'list',
-        raw,
+        raw: '',
         ordered: isordered,
         start: isordered ? +bull.slice(0, -1) : '',
         loose: false,
         items: []
       };
 
-      // Get each top-level item.
-      const itemMatch = cap[0].match(this.rules.block.item);
-
-      let next = false,
-        item,
-        space,
-        bcurr,
-        bnext,
-        addBack,
-        loose,
-        istask,
-        ischecked,
-        endMatch;
-
-      let l = itemMatch.length;
-      bcurr = this.rules.block.listItemStart.exec(itemMatch[0]);
-      for (let i = 0; i < l; i++) {
-        item = itemMatch[i];
-        raw = item;
-
-        if (!this.options.pedantic) {
-          // Determine if current item contains the end of the list
-          endMatch = item.match(new RegExp('\\n\\s*\\n {0,' + (bcurr[0].length - 1) + '}\\S'));
-          if (endMatch) {
-            addBack = item.length - endMatch.index + itemMatch.slice(i + 1).join('\n').length;
-            list.raw = list.raw.substring(0, list.raw.length - addBack);
-
-            item = item.substring(0, endMatch.index);
-            raw = item;
-            l = i + 1;
-          }
+      bull = isordered ? `\\d{1,9}\\${bull.slice(-1)}` : `\\${bull}`;
+
+      if (this.options.pedantic) {
+        bull = isordered ? bull : '[*+-]';
+      }
+
+      // Get next list item
+      const itemRegex = new RegExp(`^( {0,3}${bull})((?: [^\\n]*|\\h*)(?:\\n[^\\n]*)*(?:\\n|$))`);
```

Comment:
is this supposed to be `\n`?
```suggestion
      const itemRegex = new RegExp(`^( {0,3}${bull})((?: [^\\n]*|\\n*)(?:\\n[^\\n]*)*(?:\\n|$))`);
```

Reply:
It's all horizontal whitespaces. Could probably be ` ` but I wasn't sure if tabs or `nbsp` shows up in here.

Reply:
In javascript `\h` just matches the `h` character. I don't think javascript has a horizontal whitespace character class.

Reply:
Tested in the devtools console.

![image](https://user-images.githubusercontent.com/97994/123037319-17b14680-d3b4-11eb-93e9-a47c4cf7289b.png)

Reply:
Well I'll be. I guess we can change it to a normal space then.

Reply:
Alright, I changed all instances of `\h` to ` `.

Reply:
Maybe `[ \t]` would be better?

Reply:
I seem to be remembering... Doesn't marked already convert tabs to spaces before lexing or something?

Reply:
Ah yeah, here. Is it ok to just leave it as a space then?

https://github.com/markedjs/marked/blob/c934790c3fb2e45d4cbabf96c0536ce2e84a1efd/src/Lexer.js#L109-L110

Reply:
Ya I guess. I know there is an issue about replacing tabs in code blocks so sometime in the future we will need to change it but just spaces should work for now.